lib/index.js

```diff
@@ -206,24 +206,41 @@ api.inboxes.getAll = function(actor, query, fields, options, callback) {
 
   query = query || {};
   fields = fields || {};
+  // meta is required for permission check, if meta was
+  let stripMeta = false;
+  if(Object.keys(fields).length > 0 && !fields.meta) {
+    stripMeta = true;
+    fields.meta = true;
+  }
   options = options || {};
   async.auto({
-    // need inbox access capability for all resources to use this
-    checkPermission: callback => brPermission.checkPermission(
-      actor, PERMISSIONS.LDN_INBOX_ACCESS, callback),
-    find: ['checkPermission', (results, callback) => inboxCollection.find(
-      query, fields, options).toArray(callback)]
+    find: callback => inboxCollection.find(query, fields, options)
+      .toArray(callback),
+    // check to make sure the caller is allowed to access the documentType
+    // (Note: fields *must not* have excluded `documentType.id` in this
+    // case or else the look up for the permission check will fail)
+    getAuthorized: ['find', (results, callback) => async.filterSeries(
+      results.find, (record, callback) => brPermission.checkPermission(
+        actor, PERMISSIONS.LDN_INBOX_ACCESS, {
+          resource: record.meta,
+          translate: 'owner',
+          // record.meta does not include an `id` property
+          excludeId: true
+        }, err => callback(null, !err)), callback)]
   }, (err, results) => {
     if(err) {
       return callback(err);
     }
     // decode records
-    for(const record of results.find) {
+    for(const record of results.getAuthorized) {
+      if(stripMeta) {
+        delete record.meta;
+      }
       if('message' in record) {
         record.inbox = database.decode(record.inbox);
       }
     }
-    callback(null, results.find);
+    callback(null, results.getAuthorized);
```

Comment:
We have discussed that it's not a good idea to return a full record.  The getAll endpoint has previously been restricted to admin level permissions.  We are allowing the caller to specify fields to this API, do we want to restrict those fields to properties contained in `inbox` in this case?